core/src/auctions.ts

```diff
@@ -108,24 +108,29 @@ export const fetchAllInitialAuctions = async function (network: string): Promise
 
 export const fetchAllAuctions = async function (network: string): Promise<AuctionTransaction[]> {
     const auctions = await fetchAllInitialAuctions(network);
+    return await Promise.all(auctions.map(a => enrichAuction(network, a)));
+};
+
+export const enrichAuction = async function (
+    network: string,
+    auction: AuctionInitialInfo
+): Promise<AuctionTransaction> {
+    if (auction.debtDAI.toNumber() === 0) {
+        throw new Error(`Auction "${auction.id}" is already finished.`);
+    }
```

Comment:
It's actually not an error, sorry for the wrong suggestion, we need to set `isFinished` instead and then check it in the keeper after the enrichment

Reply:
Changed it :D